drakcore/drakcore/frontend/src/AnalysisLogs.js

```diff
@@ -0,0 +1,352 @@
+      if (chunk.line > lineIndex) {
+        return i - 1;
+      }
+    }
+    return markers.length - 1;
+  };
+
+  const isItemLoaded = (entry) => {
+    const chunkIdx = findChunkIndex(entry);
+    return chunks[chunkIdx] !== undefined;
+  };
+
+  const loadMoreItems = (startIndex, stopIndex) => {
+    // console.log(`Requesting from `, startIndex, " to ", stopIndex);
+    const startChunkIndex = findChunkIndex(startIndex);
+    const endChunkIndex = findChunkIndex(stopIndex);
+
+    let promises = [];
+    // console.log(`Downloading from `, startChunkIndex, " to ", endChunkIndex);
+    for (let idx = startChunkIndex; idx <= endChunkIndex; idx++) {
+      const chunk = index.markers[idx];
+      const isLastChunk = idx === index.markers.length - 1;
+      const nextChunk = index.markers[idx + 1];
+      const endOffset = isLastChunk ? null : nextChunk.offset;
+
+      const promise = queryData(chunk.offset, endOffset).then((chunk) => {
+        return [idx, chunk];
+      });
+
+      promises.push(promise);
+    }
+
+    let updatedPromise = new Promise((resolve, _) => {
+      // Register resolver
+      pending.current = resolve;
+    });
+
+    // When all chunks are loaded, update state
+    Promise.all(promises).then((results) => {
+      const newChunks = {};
+      for (const [idx, chunk] of results) {
+        newChunks[idx] = chunk;
+      }
+      setChunks({ ...chunks, ...newChunks });
+    });
+
+    return updatedPromise;
+  };
+
+  const loader = function ({ width, height }) {
+    return (
+      <InfiniteLoader
+        isItemLoaded={isItemLoaded}
+        loadMoreItems={loadMoreItems}
+        itemCount={lines}
+        minimumBatchSize={2000}
+      >
+        {({ onItemsRendered, ref }) => (
+          <List
+            height={height}
+            width={width}
+            itemCount={lines}
+            itemSize={30}
+            onItemsRendered={onItemsRendered}
+            ref={ref}
+          >
+            {(props) => {
+              const lineIndex = props.index;
+
+              const chunkIndex = findChunkIndex(lineIndex);
+              const chunk = chunks[chunkIndex];
+              if (chunk === undefined) {
+                return children({
+                  entry: undefined,
+                  index: lineIndex,
+                  style: props.style,
+                });
+              }
+              const lineOffset = index ? index.markers[chunkIndex].line : 0;
+              const entry = chunk[lineIndex - lineOffset];
+              return children({
+                entry,
+                index: lineIndex,
+                style: props.style,
+              });
+            }}
+          </List>
+        )}
+      </InfiniteLoader>
+    );
+  };
+  return <AutoSizer>{loader}</AutoSizer>;
+}
+
+const SERVICE_LOGS = ["drakrun.log", "drak-postprocess.log"];
+
+const DrakvufRow = ({ entry, style }) => {
+  let content;
+  let displayed = { ...entry };
+  if (entry === undefined) {
```

Comment:
Pretty sure the correct way to check this is `if (typeof entry === "undefined")`